test/unit/math/test_ad.hpp

```diff
@@ -1724,6 +1774,75 @@ void expect_ad_matvar_v(const ad_tolerances& tols, const F& f,
   test_matvar_gradient(tols, A_mv_f, A_vm_f, A_mv, A_vm);
 }
 
+/**
+ * For an unary function check that an Eigen matrix of vars and a var with an
+ * inner eigen type return the same values and adjoints to within
+ * the given tolerances. This is done by calling the function on both types,
+ * summing the results, then taking the gradient of the sum which will
+ * propogate the adjoints upwards.
+ *
+ * @tparam F A lambda or functor type that calls the unary function.
+ * @tparam T An Eigen matrix type
+ * @param tols tolerances for test
+ * @param f a lambda
+ * @param x An Eigen matrix.
+ */
+template <typename F, typename EigMat, require_eigen_t<EigMat>* = nullptr>
+void expect_ad_matvar_v(const ad_tolerances& tols, const F& f,
+                        const std::vector<EigMat>& x) {
+  using stan::plain_type_t;
+  using stan::math::promote_scalar_t;
+  using stan::math::var;
+  using stan::math::var_value;
+  using stan::math::test::type_name;
+  using vec_mat_var = std::vector<promote_scalar_t<var, EigMat>>;
+  using vec_var_mat = std::vector<var_value<plain_type_t<EigMat>>>;
+  vec_mat_var A_mv;
+  for (auto xi : x) {
+    A_mv.push_back(xi);
+  }
+  plain_type_t<decltype(f(A_mv))> A_mv_f;
+  vec_var_mat A_vm;
+  for (auto xi : x) {
+    A_vm.push_back(xi);
+  }
+  plain_type_t<decltype(f(A_vm))> A_vm_f;
+  // Check return type is correct
+  check_return_type(A_mv_f, A_mv);
+  check_return_type(A_vm_f, A_vm);
```

Comment:
In this case `A_mv_f` and `A_vm_f` are both going to be `std::vector<var>` types (there are no `var_value<EigenBase>` things going on here).

I think what this function is trying to do is just check that the semantics of `promote_var_matrix_t` are respected.

So I think I need to update `check_return_type` to handle `std::vector` s, but it might be as simple as deferring all the logic to `promote_var_matrix_t`. Will look at this later though. Right now these do nothing, but really they should be doing something if the functions returned `std::vector<var_value<EigenBase>>` things.

Reply:
Could you just change it so that if the input is an `std::vector` it expects the output to also be in a vector? Then just does the logic on the underlying type